src/coreclr/gc/gc.cpp

```diff
@@ -36239,6 +36248,47 @@ size_t gc_heap::estimated_reclaim (int gen_number)
     return est_gen_free;
 }
 
+bool gc_heap::is_full_compacting_gc_productive()
+{
+#ifdef USE_REGIONS
+    // If we needed to grow gen2 by extending either the end of its tail region
+    // or having to acquire more regions for gen2, then we view this as unproductive.
+    heap_segment* gen1_start_region = generation_start_segment (generation_of (max_generation - 1));
+    if (heap_segment_plan_gen_num (gen1_start_region) == max_generation)
+    {
+        dprintf (REGIONS_LOG, ("gen1 start region %Ix is now part of gen2, unproductive",
+            heap_segment_mem (gen1_start_region)));
+        return false;
+    }
+    else
+    {
+        heap_segment* gen2_tail_region = generation_tail_region (generation_of (max_generation));
+        if (heap_segment_plan_allocated (gen2_tail_region) >= heap_segment_allocated (gen2_tail_region))
+        {
+            dprintf (REGIONS_LOG, ("gen1 start region %Ix is now part of gen2, unproductive",
```

Comment:
Should this be a different message in the dprintf?

Reply:
LGTM!

Reply:
yes it should...I'll change it.